modules/op-defining-and-adding-triggers.adoc

```diff
@@ -0,0 +1,170 @@
+* _TriggerBindings_ extracts the fields from an event payload and stores them as parameters.
+* _TriggerTemplates_ specify how to use the parameterized data from the TriggerBindings. It defines a resource template which takes as input the trigger bindings, and performs a series of actions that result in creation of new PipelineResources and starting of a new PipelineRun.
+
+To sum up, EventListeners tie the concept of TriggerBindings and TriggerTemplates together. It listens for the incoming event, handles basic filtering using Interceptors, extracts data using TriggerBindings, and processes this data to create Kubernetes resources using TriggerTemplates.
+
+image::op-triggers.png[]
+
+
+== Procedure
+After you have assembled and started the Pipeline for the application, add TriggerTemplates, TriggerBindings, and EventListener to capture GitHub events.
+
+. Copy the content of the following sample `TriggerBinding` YAML file and save it.
++
+----
+apiVersion: tekton.dev/v1alpha1
+kind: TriggerBinding
+metadata:
+  name: vote-app-pipelinebinding
+spec:
+  params:
+  - name: git-revision
+    value: $(body.head_commit.id)
+  - name: namespace
+    value: pipelines-tutorial
+  - name: git-repo-url
+    value: $(body.repository.url)
+  - name: git-repo-name
+    value: $(body.repository.name)
+----
+
+. Create the `TriggerBinding`
++
+----
+$ oc create -f <triggerbinding-yaml-file-name.yaml>
+----
++
+Alternatively, you can create `TriggerBinding` directly from the `pipelines-tutorial` Git repository:
++
+----
+$ oc create -f https://raw.githubusercontent.com/openshift/pipelines-tutorial/master/03_triggers/01_binding.yaml
+----
+
+. Copy the content of the following sample `TriggerTemplate` YAML file and save it.
++
+----
+apiVersion: tekton.dev/v1alpha1
+kind: TriggerTemplate
+metadata:
+  name: vote-app-triggertemplate
+spec:
+  params:
+  - name: git-revision
+    description: The git revision
+    default: master
+  - name: git-repo-url
+    description: The git repository url
+  - name: namespace
+    description: The namespace to create the resources
+  - name: git-repo-name
+    description: The name of the deployment to be created / patched
+  resourcetemplates:
+  - apiVersion: tekton.dev/v1alpha1
+    kind: PipelineResource
+    metadata:
+      name: source-repo-$(params.git-repo-name)-$(uid)
+      namespace: $(params.namespace)
+    spec:
+      type: git
+      params:
+      - name: revision
+        value: $(params.git-revision)
+      - name: url
+        value: $(params.git-repo-url)
+  - apiVersion: tekton.dev/v1alpha1
+    kind: PipelineResource
+    metadata:
+      name: image-source-$(params.git-repo-name)-$(uid)
+      namespace: $(params.namespace)
+    spec:
+      type: image
+      params:
+      - name: url
+        value: image-registry.openshift-image-registry.svc:5000/$(params.namespace)/$(params.git-repo-name):latest
+  - apiVersion: tekton.dev/v1alpha1
```

Comment:
We need to update those to `v1beta1`.

Reply:
@praveen4g0 - Could you please provide me with the yaml which has v1beta1 changes for the triggers yaml
cc: @vdemeester